### PR TITLE
Add Design Tokens To Spark Design System

### PR DESCRIPTION
**Pull request by afebbraro**
Reverts sparkdesignsystem/spark-design-system#3604

### CODE REVIEW

File: angular/package.json [hunk 1]
```diff
@@ -10,10 +10,11 @@
     "lint": "ng lint",
     "coverage": "jest --coverage --config ./projects/spark-angular/jest.config.js",
     "docs:json": "compodoc --disableLifeCycleHooks -p ./tsconfig.json -e json -d .",
+    "build-styles": "cd '../styles/' && node build.js",
     "set-version": "PACKAGE_VERSION=$(node -p -e \"require('./projects/spark-angular/package.json').version\") && replace '(version).*' \"version: '$PACKAGE_VERSION'\" ./projects/spark-angular/src/environment/environment.ts",
-    "sassdocmodifiers": "sassdoc '../html/**/_*.scss' --parse > ../src/data/sass-modifiers.json",
+    "sassdocmodifiers": "sassdoc '../styles/**/_*.scss' --parse > ../src/data/sass-modifiers.json",
     "prebuild:storybook": "npm run sassdocmodifiers && npm run test:generate-output && npm run docs:json",
-    "storybook": "npm run prebuild:storybook && start-storybook -p 8006 -s ../storybook-utilities/favicon",
+    "storybook": "npm run build-styles && npm run prebuild:storybook && start-storybook -p 8006 -s ../storybook-utilities/favicon",
```
Comment (afebbraro) [inline angular/package.json:17]: Add styles build cmd to ng sb and it run when we launch sb so the vars are there in place.

File: angular/projects/spark-angular/package.json [hunk 2]
```diff
@@ -18,8 +18,7 @@
     "@angular/forms": ">=7.0.0 < 9.2.0",
     "@angular/platform-browser": ">=7.0.0 < 9.2.0",
     "@angular/platform-browser-dynamic": ">=7.0.0 < 9.2.0",
-    "@angular/router": ">=7.0.0 < 9.2.0",
-    "@sparkdesignsystem/spark": "^14.0.0"
+    "@angular/router": ">=7.0.0 < 9.2.0"
```
Comment (afebbraro) [inline angular/projects/spark-angular/package.json:21]: Remove Peer dep on spark html. Will add dep on spark-styles with alpha publish.

File: angular/projects/spark-angular/schematics/README.md [hunk 3]
```diff
@@ -19,7 +19,6 @@ ng add @sparkdesignsystem/spark-angular
 This single command will:
 
 - Install `@sparkdesignsystem/spark-angular` as a dev dependency.
-- Install `@sparkdesignsystem/spark` as a dev dependency.
```
Comment (afebbraro) [inline angular/projects/spark-angular/schematics/README.md:22]: Update Angular Schematics to remove spark install dep.

File: angular/projects/spark-angular/schematics/ng-add/index.ts [hunk 4]
```diff
@@ -31,19 +36,26 @@ const browserAnimationsModuleImport = '@angular/platform-browser/animations';
...
 const addSparkDependencies = (_options: Schema): Rule => {
   return (tree: Tree, context: SchematicContext) => {
     // Schematics automatically adds to "dependencies" but it should be "devDependencies"
     removePackageJsonDependency(tree, '@sparkdesignsystem/spark-angular');
 
     const dependencies: NodeDependency[] = [
-      { name: '@sparkdesignsystem/spark', overwrite: true, type: NodeDependencyType.Dev, version: '^13.1.0' },
-      { name: '@sparkdesignsystem/spark-angular', overwrite: true, type: NodeDependencyType.Dev, version: '^10.1.0' },
+      {
+        name: '@sparkdesignsystem/spark-angular',
```
Comment (afebbraro) [inline angular/projects/spark-angular/schematics/ng-add/index.ts:49]: Remove code to install spark html/

File: src/scss/main.scss [hunk 5]
```diff
@@ -1,6 +1,6 @@
 @import './settings';
 @import './spark-theme';
-@import '~@sparkdesignsystem/spark/spark.scss';
+@import '../../styles/spark';
```
Comment (afebbraro) [inline src/scss/main.scss:3]: will change this gatsby styles import to alpha when published

File: storybook-utilities/storybook-theming/_docs.scss [hunk 6]
```diff
@@ -1,5 +1,5 @@
-@import '../html/settings/colors';
-@import '../html/settings/settings';
+@import '../../styles/web/colors';
```
Comment (afebbraro) [inline storybook-utilities/storybook-theming/_docs.scss:1]: change sb path to import sass now from generated colors and settings files

File: storybook-utilities/storybook-theming/scss/_spark-angular-manifest.scss [hunk 7]
```diff
@@ -1,2 +1 @@
-@import '../../../html/spark.scss';
-@import '../../../html/spark-angular.scss';
+@import '../../../styles/spark.scss';
```
Comment (afebbraro) [inline storybook-utilities/storybook-theming/scss/_spark-angular-manifest.scss:1]: I moved spark-angular sass into spark.scss so we dont need to individually import angular sass anymore.

File: styles/base/inputs/_text-input.scss [hunk 8]
```diff
@@ -97,3 +96,11 @@
   font-family: $sprk-placeholder-font-family;
   color: $sprk-placeholder-color;
 }
+
+/* stylelint-disable selector-class-pattern */
+.sprk-b-TextInput.ng-invalid:not(.ng-pristine)
```
Comment (afebbraro) [inline styles/base/inputs/_text-input.scss:101]: Here is the addition of the sass from spark-angular sass.

File: styles/build.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+StyleDictionary.registerFormat({
+  name: 'themable-scss/web',
+  formatter: (tokens) => {
+    const formattedTokens = tokens.allProperties.map((token) => {
+      let tokenName = token.name;
+      if (tokenName === 'sprk-masthead-translate-y')
+        tokenName = 'sprk-masthead-translateY';
+      return `/// ${token.comment} \n$${tokenName}: ${token.value}${
+        token.themable ? ' !default' : ''
```
Comment (afebbraro) [inline styles/build.js:11]: Add check for if sass var is themable. If so, add !default.

File: styles/build.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    return formattedTokens.join('\n');
+  },
+});
+
+StyleDictionary.registerTransform({
+  name: 'size/px-to-rem',
+  type: 'value',
+  matcher(token) {
+    return (
+      token.attributes.category === 'size' &&
```
Comment (afebbraro) [inline styles/build.js:23]: have to convert all our px values to rems before the rem to pt conversion happens for ios

File: styles/components/_masthead.scss [hunk 11]
```diff
@@ -431,3 +431,12 @@
     background-color: $sprk-accordion-active-background-color;
   }
 }
+
+/* stylelint-disable selector-class-pattern */
```
Comment (afebbraro) [inline styles/components/_masthead.scss:435]: Addition of styles from spark-angular into regular spark styles

**Review by RCopeland — APPROVED** [review]
i think we'll need to figure out a way to deploy the generated code to cdn rather than commit it, but otherwise lgtm

File: styles/build.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+  formatter: (tokens) => {
+    return tokens.allProperties
+      .map((token) => {
+        let tokenName = token.name;
+        let tokenValue = token.original.value;
+        const tokenGroup = token.group;
+        const tokenType = token.type;
+
+        if (tokenName === 'sprk-masthead-translate-y')
+          tokenName = 'sprk-masthead-translateY';
```
Comment (bobwise) [inline styles/build.js:14]: y did we have to do this?

File: styles/build.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+        /*
+         * Check if the value includes a reference
+         * so that we can format the ref to a Sass Var.
+         */
+        const regex = /\{(.*?)\}/;
+
+        /*
+         * Formats a reference back into a scss var.
+         * black.value -> $sprk-black
+         */
```
Comment (bobwise) [inline styles/build.js:25]: Great job on the comments in this section!

File: styles/build.js (hunk 10 above)
Comment (bobwise) [reply styles/build.js:58]: Can we put this in a comment in the code?

Comment (afebbraro) [reply styles/build.js:58]: yeah thats a great idea!

File: styles/build.js (hunk 12 above)
Comment (afebbraro) [reply styles/build.js:14]: @bobwise, we have that one variable that has a weird naming and thus gets translated incorrectly. I'll add a comment above it in the code explaining.

**Pull request merged**